token-erc-721/chaincode-java/src/main/java/org/hyperledger/fabric/samples/erc721/ERC721TokenContract.java

```diff
@@ -0,0 +1,430 @@
+   * @param operator: operator The new approved client
+   * @param tokenId: tokenId the non-fungible token to approve
+   */
+  @Transaction(intent = Transaction.TYPE.SUBMIT)
+  public void approve(final Context ctx, final String operator, final String tokenId) {
+    ChaincodeStub stub = ctx.getStub();
+    final String sender = ctx.getClientIdentity().getId();
+    NFT nft = this.readNft(ctx, tokenId);
+    String owner = nft.getOwner();
+    boolean operatorApproval = this.isApprovedForAll(ctx, owner, sender);
+    if ((!owner.equalsIgnoreCase(sender)) && (!operatorApproval)) {
+      final String errorMessage = String.format(
+          "The sender %s is not the current owner nor an authorized operator of the token %s.",
+          sender, tokenId);
+      throw new ChaincodeException(errorMessage, ContractErrors.UNOTHERIZED_SENDER.toString());
+    }
+
+    nft.setApproved(operator);
+    CompositeKey nftKey = stub.createCompositeKey(ContractConstants.NFT.getValue(), tokenId);
+    stub.putStringState(nftKey.toString(), nft.toJSONString());
+  }
+
+  /**
+   * Enables or disables approval for a third party ("operator") to manage all the message sender's
+   * assets.
+   * 
+   * @param ctx the transaction context
+   * @param operator A client to add to the set of authorized operators
+   * @param approved: True if the operator is approved, false to revoke approval
+   */
+  @Transaction(intent = Transaction.TYPE.SUBMIT)
+  public void setApprovalForAll(final Context ctx, final String operator, final boolean approved) {
+    final String sender = ctx.getClientIdentity().getId();
+    ChaincodeStub stub = ctx.getStub();
+    final Approval nftApproval = new Approval(sender, operator, approved);
+    CompositeKey approvalKey = stub.createCompositeKey(ContractConstants.APPROVAL.getValue(), sender,
+        operator);
+    stub.putStringState(approvalKey.toString(), nftApproval.toJSONString());
+    stub.setEvent(ContractConstants.APPROVE_FOR_ALL.getValue(),
+        nftApproval.toJSONString().getBytes(UTF_8));
+
+  }
+
+  /**
+   * Get the approved client for a single non-fungible token.
+   * 
+   * @param ctx the transaction context
+   * @param tokenId The non-fungible token to find the approved client for
+   * @return Return The approved client for this non-fungible token, or null if there is none
+   */
+  @Transaction(intent = Transaction.TYPE.EVALUATE)
+  public String getApproved(final Context ctx, final String tokenId) {
+    NFT nft = this.readNft(ctx, tokenId);
+    return nft.getApproved();
+  }
+
+  /**
+   * Transfers the ownership of a non-fungible token from one owner to another owner.
+   * 
+   * @param ctx the transaction context
+   * @param from the current owner of the non-fungible token
+   * @param to the new token owner
+   * @param tokenId The non-fungible token to transfer
+   */
+  @Transaction(intent = Transaction.TYPE.SUBMIT)
+  public void transferFrom(final Context ctx, final String from, final String to,

```

Comment:
can ERC721 based token be transferred?

Reply:
The link for ERC721 standard is  https://eips.ethereum.org/EIPS/eip-721. 
The details of transferFrom is given below.
 /// @notice Transfer ownership of an NFT -- THE CALLER IS RESPONSIBLE
    ///  TO CONFIRM THAT `_to` IS CAPABLE OF RECEIVING NFTS OR ELSE
    ///  THEY MAY BE PERMANENTLY LOST
    /// @dev Throws unless `msg.sender` is the current owner, an authorized
    ///  operator, or the approved address for this NFT. Throws if `_from` is
    ///  not the current owner. Throws if `_to` is the zero address. Throws if
    ///  `_tokenId` is not a valid NFT.
    /// @param _from The current owner of the NFT
    /// @param _to The new owner
    /// @param _tokenId The NFT to transfer
    function transferFrom(address _from, address _to, uint256 _tokenId) external payable;